molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/index/ElasticsearchIndexCreator.java

```diff
@@ -1,97 +1,81 @@
 	{
-		// Wait until elasticsearch is ready
-		client.admin().cluster().prepareHealth().setWaitForYellowStatus().execute().actionGet();
-		boolean hasIndex = client.admin().indices().exists(new IndicesExistsRequest(indexName)).actionGet().isExists();
-		if (!hasIndex)
+		try
+		{
+			// Wait until elasticsearch is ready
+			elasticsearchUtils.waitForYellowStatus();
+
+			if (!elasticsearchUtils.indexExists(indexName))
+			{
+				createIndexInternal(indexName);
+			}
+		}
+		catch (IOException e)
 		{
-			createIndexInternal(indexName);
+			throw new UncheckedIOException(e);
 		}
 	}
 
-	public void createIndex(String indexName) throws IOException
+	private void createIndexInternal(String indexName) throws IOException
 	{
-		// Wait until elasticsearch is ready
-		client.admin().cluster().prepareHealth().setWaitForYellowStatus().execute().actionGet();
-		
-		createIndexInternal(indexName);
-	}
-	
-	private void createIndexInternal(String indexName) throws IOException {
 		if (LOG.isTraceEnabled()) LOG.trace("Creating Elasticsearch index [" + indexName + "] ...");
-		Builder settings = ImmutableSettings.settingsBuilder().loadFromSource(XContentFactory.jsonBuilder()
-                .startObject()
-                    .startObject("analysis")
-                        .startObject("analyzer")
-                            .startObject(DEFAULT_ANALYZER)
-                                .field("type", "custom")
-                                .field("filter", new String[]{"lowercase", DEFAULT_STEMMER})
-                                .field("tokenizer", DEFAULT_TOKENIZER)
-                                .field("char_filter", "html_strip")
-                            .endObject()
-                            .startObject(NGRAM_ANALYZER)
-                                .field("type", "custom")
-                                .field("filter", new String[]{"lowercase"})
-                                .field("tokenizer", NGRAM_TOKENIZER)
-                            .endObject()
-                        .endObject()
-	                    .startObject("filter")
-	                    	.startObject(DEFAULT_STEMMER)
-	                    		.field("type", "stemmer")
-	                    		.field("name", "english")
-	                    	.endObject()
-	                    .endObject()
-	                    .startObject("tokenizer")
-	                    	.startObject(DEFAULT_TOKENIZER)
-	                    		.field("type", "pattern")
-	                    		.field("pattern", "([\\W\\.]+)")
-	                    	.endObject()
-	                    	.startObject(NGRAM_TOKENIZER)
-	                    		.field("type", "nGram")
-	                    		.field("min_gram", 1)
-	                    		.field("max_gram", 10)
-	                    	.endObject()
-	                    .endObject()
-                    .endObject()
-                .endObject().string());
+		Builder settings = ImmutableSettings.settingsBuilder().loadFromSource(
```

Comment:
Please preserve formatting of settings builder, you can do this in Eclipse by performing undo directly after save.